### ISSUE TITLE
Proposal - Move @position from Rake::Command to CommandFormatter

### ISSUE THREAD

**Issue by robd** [mattbrictson/airbrussh#49]
As a result of #48, which removes `first_execution?` from `Rake::Command` I wonder if  `Rake::Command` is no longer holding it's weight. I think perhaps we should move the `@position` attribute from the `Rake::Command` on to the `CommandFormatter` and delete the `Rake::Command` class. I think this is a simple change which could be achieved like this:

```Ruby
# Context
# Replace decorate_command with a position method:
def position
  history.index(command.to_s)
end

# ConsoleFormatter
# Pass the position to the CommandFormatter as a constructor arg in decorate:
def decorate(command)
  Airbrussh::CommandFormatter.new(command, @context.position(command))
end

# CommandFormatter
# Change the one use of position to use the new attribute:
def number
  format("%02d", @position + 1)
end
```

The upside of this change is that we no longer have to double wrap the `Command`, and it makes it easier to see where the `@position` is coming from on the `CommandFormatter`. Since the `position` is only used on the `CommandFormatter` I think it makes sense for it to be an attribute on there.

@mattbrictson Do you think this would be an improvement? Am I missing anything here?

**Comment by mattbrictson** [comment]
Yes, this looks like a great improvement! :+1:

**Comment by mattbrictson** [comment]
Closed via #50.
